### ISSUE TITLE
Generate ObjC source for .strings files.

### ISSUE THREAD

**Issue by mr-fixit** [SwiftGen/SwiftGen#376]
I wrote a couple of templates, and would like to contribute them back.  I'm opening this issue as suggested by `CONTRIBUTING.md`.

Given the following in `Localizable.strings`
```"cmd.delete" = "Delete";```

... the following is generated:

```
// Localizable.h
@import Foundation;
NS_ASSUME_NONNULL_BEGIN

@interface Localized : NSObject
  + (NSString*)cmdDelete; // cmd.delete --> "Delete"
@end
```

and 

```
// Localizable.m
#import "Localizable.h"

#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wformat-security"

NSString* tr(NSString* key, ...) {
    va_list args;
    va_start(args, key);

    NSString *format = [NSBundle.mainBundle localizedStringForKey:key value:@"" table:nil];
    NSString *result = [[NSString alloc] initWithFormat:format arguments:args];
    return result;
};

@implementation Localized : NSObject
+ (NSString*)cmdDelete {  
    return tr(@"cmd.delete"); // Delete
}
@end```

**Comment by AliSoftware** [comment]
👍 looks like a nice interface too me 👌

Iirc I think there's a way to annotate functions like `tr` to tell the compiler they behave like `printf`, would be nice to use that annotation here, I'll try to find how it looks

**Comment by AliSoftware** [comment]
Ha, here it is: [NSHipster article](http://nshipster.com/__attribute__/)

Which makes me realize that you can forget about what I said, that attribute won't be useful in the current case, because it's used by the compiler to check if the variadic parameters passed match the placeholders in the "format" parameter… but in the case of `tr` we don't pass a format parameter with placeholders, we pass a key… so that attribute wouldn't make sense here, nvm.

**Comment by djbe** [comment]
What we can still do is generate functions with the correct signatures for parameter types. I have no idea what to name the parameters though? p1, p2, pN?
```objective-c
@interface Localized: NSObject
  + (NSString*)cmdDeleteWithP1: (NSObject *)a1 p2: (float)a2;
@end
```

As @AliSoftware said, you can just use varargs, but will have no way of checking the arguments at compile time.

**Comment by AliSoftware** [comment]
Fun fact: in Objective-C, parameters are not _required_ to have a name.

You can totally do `+ (NSString*)cmdDelete: (id)p1, :(float)p2, :(int)p3;`
And user will call it `[Localized cmdDelete :obj, :5.1, :42`.

Granted, it's very unusual (I know only one function in Apple's SDK with such signature: [CAMediaTiming init](https://developer.apple.com/documentation/quartzcore/camediatimingfunction?language=objc)) and it can feel very weird, but for our special case it's good to know that it's possible and can make us wonder if it wouldn't be a good fit here.

**Comment by djbe** [comment]
Woah, TIL. Yep, that totally fits with what we'd want.

**Comment by mr-fixit** [comment]
My current implementation treats`%@` in format string as an `NSString*` parameter, and a `%d` as an `NSInteger`.

Example:
```
"general.num.categories" = "%d Category(s)";
"foo.at.bar" = "%@ @ %@";
```

becomes

```
+ (NSString*)generalNumCategories:(NSInteger)p1;       // %d Category(s)
+ (NSString*)fooAtBar:(NSString*)p1 p1:(NSString*)p2;       // %@ @ %@
```

The parameter naming is messed up (note that `p1` is the name of the first parameter, but also the tag of the second.  We could drop the name entirely (nice trick), or to use Stencil's `counter0` to fix the following:
```{% macro parametersBlock types %}{% filter removeNewlines:"leading" %}
  {% for type in types %}
    ({% call paramTranslate type %})p{{ forloop.counter }}{% if not forloop.last %} p{{ forloop.counter }}:{% endif %}
  {% endfor %}
{% endfilter %}{% endmacro %}```

**Comment by djbe** [comment]
Actually, `%@` is for:
> Objective-C object, printed as the string returned by descriptionWithLocale: if available, or description otherwise. Also works with CFTypeRef objects, returning the result of the CFCopyDescription function.

That's why my example mentioned `NSObject`. Maybe it can be `id` or something similar, my Objective-C knowledge has degraded quite a bit 😊

**Comment by AliSoftware** [comment]
If you don't want to drop the parameter external name (the one before the `:`) entirely if you feel it's odd, maybe we can instead use the same external name for all parameters?

`+ (NSString*) cmdDelete:(id)p1 and:(float)p2 and:(int)p3;`

Not sure `and:` feels _that_ great, if you have any better idea…

**Comment by mr-fixit** [comment]
You're right, I'll switch to `NSObject*` or `id`.  I'll put up a PR so you guys can have something to look at.

**Comment by mr-fixit** [comment]
addressed in https://github.com/SwiftGen/SwiftGen/pull/378

**Comment by AliSoftware** [comment]
Fixed in #378 🎉
